pkgs/by-name/no/nolibc/package.nix

```diff
@@ -0,0 +1,66 @@
+{
+  lib,
+  stdenvNoLibc,
+  linuxHeaders,
+}:
+
+stdenvNoLibc.mkDerivation (finalAttrs: {
+  pname = "nolibc";
+
+  inherit (linuxHeaders) version src;
+
+  # Nolibc depends on these headers
+  propagatedBuildInputs = [ linuxHeaders ];
+
+  setSourceRoot = ''
+    sourceRoot=$(echo */tools/include/nolibc)
+  '';
+
+  makeFlags = [ "ARCH=${stdenvNoLibc.hostPlatform.linuxArch}" ];
+  makeTarget = [ "headers" ];
+
+  # The Makefile insists on installing to $(OUTPUT)/sysroot, so we satisfy
+  # their wish, use the default of current directory, and copy in installPhase.
+  installPhase = ''
+    runHook preInstall
+    mkdir -p "$out"
+    cp -r -t "$out" sysroot/*
+    runHook postInstall
+  '';
+
+  # {check,installCheck}Phase are disabled if cross, so we use our own phase
+  postPhases = [ "nolibcCheckPhase" ];
+
+  # For nolibcCheckPhase
+  hardeningDisable = [
+    "stackprotector" # Segfaults due to no TLS support
+  ];
+
+  nolibcCheckPhase =
+    ''
+      echo "Doing a compilation test"
+
+      cat > hello-world.c <<END
+      #include <stdio.h>
+
+      int main() { puts("Hello, world!"); }
```

Comment:
Maybe we should return 0 explicitly, or does the nolibc as no mechanicm to handle this?

Reply:
https://en.cppreference.com/w/c/language/main_function

> The main function has several special properties:
>
> [...]
>
> If the return type is compatible with `int` and control reaches the terminating `}`, the value returned to the environment is the same as if executing `return 0;`. (Since C99)

Reply:
Oh wait in this sentence does it imply the return type being the one of `puts`?

Reply:
No, the return type is `int` as written in `int main()`